src/StateMaps/PDEConstrainedFunctionals.jl

```diff
@@ -428,3 +428,127 @@ function Base.show(io::IO,::MIME"text/plain",f::AbstractPDEConstrainedFunctional
   print(io,"$(nameof(typeof(f))):
     num_constraints: $N")
 end
+
+struct CustomPDEConstrainedFunctionals{N,A} <:  AbstractPDEConstrainedFunctionals{N}
+  φ_to_jc :: Function
+  dJ :: Vector{Float64}

```

Comment:
Once this is merged, I need to test this in parallel and see if Zygote's jacobian function will work there. I suspect not.